crates/autopilot/src/run_loop.rs

```diff
@@ -134,9 +140,30 @@ impl RunLoop {
         let mut last_auction = None;
         let mut last_block = None;
         let self_arc = Arc::new(self);
+        let mut leader = self_arc
+            .persistence
+            .leader("autopilot_startup".to_string())
+            .await;
+
         loop {
+            let is_leader = leader.tick().await.unwrap_or_else(|err| {
+                tracing::warn!(error=%err, "failed to get leader");
+                false
+            });
+            let start_block = if is_leader {
+                self_arc.update_caches(&mut last_block, true).await
+            } else {
+                self_arc.update_caches(&mut last_block, false).await;
+                continue;
+            };
+
+            // caches are warmed up, we're ready to do leader work
+            if let Some(readiness) = self_arc.readiness.as_ref() {
+                readiness.store(true, Ordering::Release);
+            }
+
             let auction = self_arc
```

Comment:
I think here (lines 165-173) we'd ideally have a `select!` macro over this auction cuttinag and a sigterm handler.